core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala

```diff
@@ -244,163 +251,115 @@ sealed abstract class BitcoinTxBuilder extends TxBuilder {
       unsignedTx: Transaction,
       dummySignatures: Boolean)(
       implicit ec: ExecutionContext): Future[Transaction] = {
-    val outpoint = utxo.outPoint
-    val output = utxo.output
-    val signers = utxo.signers
-    val redeemScriptOpt = utxo.redeemScriptOpt
-    val scriptWitnessOpt = utxo.scriptWitnessOpt
-    val hashType = utxo.hashType
     val idx =
-      unsignedTx.inputs.zipWithIndex.find(_._1.previousOutput == outpoint)
+      unsignedTx.inputs.zipWithIndex.find(_._1.previousOutput == utxo.outPoint)
     if (idx.isEmpty) {
       Future.fromTry(TxBuilderError.MissingOutPoint)
     } else {
       val inputIndex = UInt32(idx.get._2)
       val oldInput = unsignedTx.inputs(inputIndex.toInt)
 
-      output.scriptPubKey match {
-        case _: P2PKScriptPubKey =>
-          P2PKSigner
-            .sign(utxo, unsignedTx, dummySignatures)
-            .map(_.transaction)
-        case _: P2PKHScriptPubKey =>
-          P2PKHSigner
-            .sign(utxo, unsignedTx, dummySignatures)
-            .map(_.transaction)
-        case _: MultiSignatureScriptPubKey =>
-          MultiSigSigner
-            .sign(utxo, unsignedTx, dummySignatures)
-            .map(_.transaction)
-        case _: LockTimeScriptPubKey =>
-          LockTimeSigner
+      utxo match {
+        case p2shInfo: P2SHNoNestSpendingInfo =>
+          signAndAddInputForP2SH(p2shInfo,
+                                 unsignedTx,
+                                 oldInput.sequence,
+                                 inputIndex.toInt,
+                                 dummySignatures)
+        case p2shNestedSegwitInfo @ P2SHNestedSegwitV0UTXOSpendingInfo(
+              _,
+              amount,
+              scriptPubKey,
+              _,
+              hashType,
+              redeemWitnessScript,
+              _) =>
+          redeemWitnessScript match {
+            case _: P2WSHWitnessSPKV0 =>
+              signAndAddInputForP2SH(p2shNestedSegwitInfo,
```

Comment:
Don't we need to call 

```scala
val uwtx = WitnessTransaction.toWitnessTx(unsignedTx)
```

here?

Reply:
Actually it doesn't look like it, with the p2sh hack where we replace the output script?

Reply:
Yeah, shouldn't need to do anything special since p2sh recurses on this external function